src/main/java/gg/projecteden/nexus/models/vaults/VaultUser.java

```diff
@@ -0,0 +1,69 @@
+package gg.projecteden.nexus.models.vaults;
+
+import dev.morphia.annotations.Converters;
+import dev.morphia.annotations.Entity;
+import dev.morphia.annotations.Id;
+import gg.projecteden.mongodb.serializers.UUIDConverter;
+import gg.projecteden.nexus.framework.exceptions.postconfigured.InvalidInputException;
+import gg.projecteden.nexus.framework.interfaces.PlayerOwnedObject;
+import gg.projecteden.nexus.framework.persistence.serializer.mongodb.ItemStackConverter;
+import gg.projecteden.nexus.utils.PlayerUtils;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static gg.projecteden.utils.StringUtils.plural;
+
+@Data
+@Entity(value = "vault_user", noClassnameStored = true)
+@NoArgsConstructor
+@AllArgsConstructor
+@RequiredArgsConstructor
+@Converters({UUIDConverter.class, ItemStackConverter.class})
+public class VaultUser implements PlayerOwnedObject {
+	@Id
+	@NonNull
+	private UUID uuid;
+	private Map<Integer, List<ItemStack>> vaults = new HashMap<>();
+	private int limit;
+
+	public static final int ROWS = 4;
+
+	public List<ItemStack> get(int page) {
+		return get(page, getOnlinePlayer());
+	}
+
+	public List<ItemStack> get(int page, Player viewer) {
+		if (!getRank().isAdmin())
+			if (page > limit) {
+				final String descriptor = PlayerUtils.isSelf(this, viewer) ? "You only own" : getNickname() + " only owns";
+				throw new InvalidInputException(descriptor + " &e" + limit + plural(" &cvault", limit));
+			}
+
+		return vaults.computeIfAbsent(page, $ -> Collections.nCopies(ROWS * 9, new ItemStack(Material.AIR)));
```

Comment:
using an `ItemStack` for the second parameter here instead of a `Supplier<ItemStack>` seems, er, suspicious. having a list filled with 36 references to the same object in memory seems ripe for potential issues

Reply:
I had the same thought, however I didnt know of a method that would accept a supplier. I decided to try it and saw no issues. But I'll change it.

Reply:
Changed to simply `new ArrayList<>()` as the menu doesnt need n rows of air, it'll do it itself